src/matches.rs

```diff
@@ -112,34 +112,40 @@ impl LateLintPass for MatchPass {
 fn check_single_match(cx: &LateContext, ex: &Expr, arms: &[Arm], expr: &Expr) {
     if arms.len() == 2 &&
        arms[0].pats.len() == 1 && arms[0].guard.is_none() &&
-       arms[1].pats.len() == 1 && arms[1].guard.is_none() &&
-       is_unit_expr(&arms[1].body) {
+       arms[1].pats.len() == 1 && arms[1].guard.is_none() {
```

Comment:
The conclusion is: add a lint that suggests turning regular `if/else` into a match on a boolean xD

One thing I have noticed is that I find statements much more readable with `if let` and expressions with `match`